AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -47,11 +70,27 @@ public struct GraphQLFilterConverter {
 /// Extension to translate a `QueryPredicate` into a GraphQL query variables object
 extension QueryPredicate {
 
+    /// - Warning: Although this has `public` access, it is intended for internal use and should not be used directly
+    ///   by host applications. The behavior of this may change without warning.
+    public func graphQLFilter(for modelSchema: ModelSchema?) -> GraphQLFilter {
+        if let operation = self as? QueryPredicateOperation {
+            return operation.graphQLFilter(for: modelSchema)
+        } else if let group = self as? QueryPredicateGroup {
+            return group.graphQLFilter(for: modelSchema)
+        }
+
+        preconditionFailure(
+            "Could not find QueryPredicateOperation or QueryPredicateGroup for \(String(describing: self))")
```

Comment:
no action needed; just a thought exercise here; If developers use `QueryPredicateConstant.all` then it will fail. Is there a way for developers to get into this state? Could they somehow pass the predicate as `QueryPredicateConstant.all` and then that gets synced to the API and fail here? If so, what should API do when they see a predicate that is ".all"? 

Secondly, precondionFailure affects live apps as well. In one of my latest PR's, I've started using `assert(false, "message")` instead so that it crashes during debug builds